ledger/participant-integration-api/src/test/suite/scala/platform/store/interning/StringInterningCacheSpec.scala

```diff
@@ -0,0 +1,216 @@
+    val testee = new PrefixingStringInterningCache((_, _) => _ => Future.successful(Nil))
+    partyShouldNotPresent(testee, "p1")
+    partyShouldNotPresent(testee, "p2")
+    partyShouldNotPresent(testee, "22:same:name")
+    templateShouldNotPresent(testee, "22:t:a")
+    templateShouldNotPresent(testee, "22:t:b")
+    templateShouldNotPresent(testee, "22:same:name")
+    testee.add(
+      new DomainStringIterators(
+        parties = List("p1", "p2", "22:same:name").iterator,
+        templateIds = List("22:t:a").iterator,
+      )
+    ) shouldBe Vector(
+      1 -> "p|p1",
+      2 -> "p|p2",
+      3 -> "p|22:same:name",
+      4 -> "t|22:t:a",
+    )
+    partyShouldPresent(testee, "p1", 1)
+    partyShouldPresent(testee, "p2", 2)
+    partyShouldPresent(testee, "22:same:name", 3)
+    partyShouldNotPresent(testee, "unknown")
+    templateShouldPresent(testee, "22:t:a", 4)
+    templateShouldNotPresent(testee, "22:t:b")
+    templateShouldNotPresent(testee, "22:same:name")
+    templateShouldNotPresent(testee, "22:unkno:wn")
+    testee.add(
+      new DomainStringIterators(
+        parties = List("p1", "p2").iterator,
+        templateIds = List("22:t:a", "22:t:b", "22:same:name").iterator,
+      )
+    ) shouldBe Vector(
+      5 -> "t|22:t:b",
+      6 -> "t|22:same:name",
+    )
+    partyShouldPresent(testee, "p1", 1)
+    partyShouldPresent(testee, "p2", 2)
+    partyShouldPresent(testee, "22:same:name", 3)
+    partyShouldNotPresent(testee, "unknown")
+    templateShouldPresent(testee, "22:t:a", 4)
+    templateShouldPresent(testee, "22:t:b", 5)
+    templateShouldPresent(testee, "22:same:name", 6)
+    templateShouldNotPresent(testee, "22:unkno:wn")
+  }
+
+  it should "not update cache if last id is behind" in {
+    val testee = new PrefixingStringInterningCache((from, to) =>
+      _ => {
+        from shouldBe 0
+        to shouldBe 6
+        Future.successful(
+          Vector(
+            1 -> "p|p1",
+            2 -> "p|p2",
+            3 -> "p|22:same:name",
+            4 -> "t|22:t:a",
+            5 -> "t|22:t:b",
+            6 -> "t|22:same:name",
+          )
+        )
+      }
+    )
+    partyShouldNotPresent(testee, "p1")
+    partyShouldNotPresent(testee, "p2")
+    partyShouldNotPresent(testee, "22:same:name")
+    templateShouldNotPresent(testee, "22:t:a")
+    templateShouldNotPresent(testee, "22:t:b")
+    templateShouldNotPresent(testee, "22:same:name")
+    testee.updateStringInterningCache(6).map { _ =>
+      partyShouldPresent(testee, "p1", 1)
+      partyShouldPresent(testee, "p2", 2)
+      partyShouldPresent(testee, "22:same:name", 3)
+      partyShouldNotPresent(testee, "unknown")
+      templateShouldPresent(testee, "22:t:a", 4)
+      templateShouldPresent(testee, "22:t:b", 5)
+      templateShouldPresent(testee, "22:same:name", 6)
+      templateShouldNotPresent(testee, "22:unk:nown")
+    }
+  }
+
+  it should "be able to update working cache correctly" in {
```

Comment:
Nitpicking here: these tests can be more concise.
e.g. why is this one asserting synchronously inserting 2 elements and updating another 4? Why not 1 and 1?

Reply:
respectively: 
* to test both domains are updated, going not from scratch
* to test update is hitting both domains with colliding strings